src/charm.py

```diff
@@ -675,6 +845,38 @@ def _routing_mode(self) -> _RoutingMode:
         """
         return _RoutingMode(self.config["routing_mode"])
 
+    @property
+    def version(self) -> Optional[str]:
+        """Return the workload version."""
+        if not self.container.can_connect():
+            return None
+
+        version_output, _ = self.container.exec([self._bin_path, "version"]).wait_output()
+        # Output looks like this:
+        # Version:      2.9.6
+        # Codename:     banon
+        # Go version:   go1.18.9
+        # Built:        2022-12-07_04:28:37PM
+        # OS/Arch:      linux/amd64
+
+        if result := re.search(r"Version:\s*(\d*\.\d*\.\d*)", version_output):
```

Comment:
`re.search(r'Version:\s+((\d+,?){3,})')`

Reply:
I actually fat-fingered the suggestion, which should have been:

```python
re.search(r'Version:\s+((\d+\.?){3,})') # note: \. instead of , 
```

But, nominally, these will also match the current regexp:

```
Version:123%456$789^
Version:     123456$7%9#
```

If we don't care how many groups, it could be `...((\d+\.?)+)`. If we want at least 3, then `{3,}`. Other than that, we probably _do_ to make sure that it's `..\s+((\d+\.?){1,}...)` or `..\s+((\d+\.?)+...)`as "at least one whitespace character, followed by at least one block of numbers.

If we really, really wanted to future proof it, just make it:

```python
if result := re.search(r'Version:\s+(?P<version>(\d+\.?){0,}(\d+$))', version_output):
  return re.groupdict()["version"]
```

There's some examples of potentially "valid" and "invalid" matches in a tester [here](https://regex101.com/r/LaG1ey/1)

Reply:
Nice regex-foo!
I think you convinced me to change to `Version:\s*(.+)`.